user_tools/src/spark_rapids_tools/enums.py

```diff
@@ -139,8 +143,17 @@ def _missing_(cls, value):
         # convert hyphens to underscores
         value = value.replace('-', '_')
         for member in cls:
-            if member.lower() == value:
+            if member.value.lower() == value:
                 return member
+        # This supports variants like 'databricks_aws_photon' -> 'databricks_aws'
+        if '_' in value:
+            parts = value.split('_')
+            if len(parts) > 2:
+                base_value = '_'.join(parts[:-1])
+                for member in cls:
+                    if member.value.lower() == base_value:
+                        return member
+
```

Comment:
That's a tricky change because it is not explicit that we use `databricks_aws` instead of `databricks_aws_photon`. There will be no feasible way to construct the `databricks_aws_photon`.
Instead, we can make an explicit `normalization` so the sites calling it are aware of this implicit mapping.

Reply:
FWIW, qualx code has been assuming hyphens in "platforms" and underscores to delimit "variants", e.g. `databricks-aws_photon`.